server/engine/engine.go

```diff
@@ -148,6 +150,35 @@ func (engine *Engine) startDeploymentExecutions() {
 	}
 }
 
+func GetLatestExecution(db *gorm.DB, step model.Step) model.Execution {
+	latestExecution := model.Execution{}
+	// Avoid GORM error from Last() if no executions yet
+	var count int64
+	db.Model(&model.Execution{}).Where("step_id = ?", step.ID).Count(&count)
```

Comment:
If you rebase, there is a new function called countStepExecutions that you can use here.

Reply:
done